Week-4/Assignment-1/index.js

```diff
@@ -0,0 +1,116 @@
+class TusharPromise {
+    /**
+     * @constructor constructor function
+     * @param {function} cb 
+     * @property promiseChain: storing each promise from the chain
+     * @property handleError: a placeholder for the error function to catch errors
+     * @property handleFinally: a placeholder for the finally function called on finally block
+     */
+    constructor(cb) {
+        this.promiseChain = [];
+
+        this.handleError = () => { };
+        this.handleFinally = () => { };
+
+        this.resolve = this.resolve.bind(this);
+        this.reject = this.reject.bind(this);
+
+        cb(this.resolve, this.reject);
+    }
+
+    /**
+     * @function then: a thanable handler function to illustrate a Promise-like Object
+     * @param {function} successCb: a callback function which will be called on success
+     * @returns the current context
+     */
+    then(successCb) {
+        this.promiseChain.push(successCb);
+        return this;
+    }
+
+    /**
+     * @function catch: a handler function to catch the errors when called
```

Comment:
@tusharwalzade216 these comments are super useful.
Please educate me as well on writing these sometime! Am not very familiar with this lingo & conventions around it.

Reply:
Sure @GdhawanG, I'm unsure if these comments are perfect, however I'll definitely let you know some of the *documentation comments* I prefer to use.